packages/@react-spectrum/steplist/src/StepList.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames, useDOMRef, useStyleProps} from '@react-spectrum/utils';
+import {DOMRef} from '@react-types/shared';
+import {filterDOMProps} from '@react-aria/utils';
+import {getChildrenSafe} from './utils';
+import React, {Ref} from 'react';
+import {SpectrumStepListProps} from '@react-types/steplist';
+import {StepListItem} from './StepListItem';
+import styles from '@adobe/spectrum-css-temp/components/steplist/vars.css';
+import {useProviderProps} from '@react-spectrum/provider';
+import {useStepList} from '@react-aria/steplist';
+import {useStepListState} from '@react-stately/steplist';
+
+function StepList<T extends object>(props: SpectrumStepListProps<T>, ref: DOMRef) {
+  props = useProviderProps(props);
+  let {styleProps} = useStyleProps(props);
+  let state = useStepListState(props);
+  let ariaProps = useStepList(); // useStepList(props, state);
+  let domRef = useDOMRef(ref) as Ref<HTMLOListElement>;
+  const {children, isEmphasized, isReadOnly} = props;
+  // Not using React.Children.toArray because it mutates the key prop.
+  const childArray = getChildrenSafe<T>(children);
+  const stepListItems = childArray.map((child, index) => {
+    let key = child.key ?? index;
+    return (
+      <StepListItem
+        itemKey={key}
+        isCurrent={key === state.selectedKey}
+        isComplete={state.isCompleted(key)}
+        isDisabled={false}
+        isEmphasized={isEmphasized}
+        isReadOnly={isReadOnly}
+        isNavigable={state.isNavigable(key)}
+        onItemSelected={state.setSelectedKey}>
+        {child.props.children}
```

Comment:
So this isn't making use of the collection, see https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/actiongroup/src/ActionGroup.tsx#L136

Reply:
I was trying to follow the pattern of Breadcrumbs (which was given as a good example in the writeup).  https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/breadcrumbs/src/Breadcrumbs.tsx#L187
It does it (more or less) the same way as I am in this StepList

Reply:
Hm, thats curious, it seems like Breadcrumbs doesn't actually make use of any of our collections code (and thus doesn't support things like `selectedKey`), perhaps because the selected item is always the last rendered item in the list so there wasn't any need for any of the SelectionManager related hooks. However, you do use `useSingleSelectListState` in your `useStepListState` meaning that you have access to `collection` and could follow a similar pattern to `ActionGroup` that @snowystinger pointed out above. That would allow end users to provide a list of steps in a "dynamic" fashion as described [here](https://react-spectrum.adobe.com/react-stately/collections.html#dynamic-collections) and not have to do an array map instead. That would mean you could perhaps pass `state` to `useStepListItem` as well and not have to worry about rerenders happening on every StepItem when only one of them changes?

I haven't given this a go myself so not 100% on if there are any pitfalls, lemme know what you think.

Reply:
Cool, I'll give that a try.

Reply:
That worked! I added a story that uses Async Items.

Reply:
Sweeeet